Index: /home/dsh/for_report/server/bin/report/report_sxw.py
===================================================================
--- /home/dsh/for_report/server/bin/report/report_sxw.py	(revision 8211)
+++ /home/dsh/for_report/server/bin/report/report_sxw.py	(working copy)
@@ -291,8 +291,11 @@
 					return found
 		return False
 
-	def _add_header(self, node):
-		rml_head = tools.file_open('custom/corporate_rml_header.rml').read()
+	def _add_header(self, node,header_rml):
+		if header_rml:
+			rml_head = tools.file_open(header_rml).read()
+		else:
+			rml_head = tools.file_open('custom/corporate_rml_header.rml').read()
 		head_dom = xml.dom.minidom.parseString(rml_head)
 		#for frame in head_dom.getElementsByTagName('frame'):
 		#	frame.parentNode.removeChild(frame)
@@ -317,7 +320,7 @@
 		self.ids = ids
 		self.objects = objects
 
-	def _parse(self, rml_dom, objects, data, header=False):
+	def _parse(self, rml_dom, objects, data, header=False,header_rml=None):
 		self.node_context = {}
 		self.dom = rml_dom
 		self._node = self.dom.documentElement
@@ -322,7 +325,7 @@
 		self.dom = rml_dom
 		self._node = self.dom.documentElement
 		if header:
-			self._add_header(self._node)
+			self._add_header(self._node,header_rml)
 		self._parse_node()
 		res = self.dom.documentElement.toxml('utf-8')
 		return res
@@ -328,7 +331,7 @@
 		return res
 
 class report_sxw(report_rml):
-	def __init__(self, name, table, rml, parser=rml_parse, header=True):
+	def __init__(self, name, table, rml, parser=rml_parse, header=True,header_rml=None):
 		report_rml.__init__(self, name, table, rml, '')
 		self.name = name
 		self.parser = parser
@@ -333,6 +336,7 @@
 		self.name = name
 		self.parser = parser
 		self.header = header
+		self.header_rml = header_rml
 
 	def getObjects(self, cr, uid, ids, context):
 		table_obj = pooler.get_pool(cr.dbname).get(self.table)
@@ -354,10 +358,10 @@
 		rml_parser = self.parser(cr, uid, self.name2, context)
 		objs = self.getObjects(cr, uid, ids, context)
 		rml_parser.preprocess(objs, data, ids)
-		
+
 		rml_dom = xml.dom.minidom.parseString(rml)
-		
-		rml2 = rml_parser._parse(rml_dom, objs, data, header=self.header)
+
+		rml2 = rml_parser._parse(rml_dom, objs, data, header=self.header,header_rml = self.header_rml)
 		#if os.name != "nt":
 		#	f = file("/tmp/debug.rml", "w")
 		#	f.write(rml2)
